csvcubed/csvcubed/readers/configdeserialiser.py

```diff
@@ -0,0 +1,594 @@
+#             or (isinstance(maybe_code_list, bool) and maybe_code_list is True)
+#         )
+#         and maybe_parent_uri
+#         == "http://purl.org/linked-data/sdmx/2009/dimension#refPeriod"
+#         and maybe_property_value_url is not None
+#         and maybe_property_value_url.startswith("http://reference.data.gov.uk/id/")
+#     )
+#
+#     if is_date_time_code_list:
+#         column_name_csvw_safe = csvw_column_name_safe(column_label)
+#         columnar_data = pandas_input_to_columnar_str(column_data)
+#         concepts = [
+#             DuplicatedQbConcept(
+#                 existing_concept_uri=uritemplate.expand(
+#                     maybe_property_value_url, {column_name_csvw_safe: c}
+#                 ),
+#                 label=c,
+#             )
+#             for c in sorted(set(columnar_data))
+#         ]
+#         return CompositeQbCodeList(
+#             CatalogMetadata(column_label),
+#             concepts,
+#         )
+#
+#     elif maybe_code_list is not None:
+#         if isinstance(maybe_code_list, str):
+#             return ExistingQbCodeList(maybe_code_list)
+#
+#         #elif isinstance(maybe_code_list, bool) and not maybe_code_list:
+#         elif isinstance(maybe_code_list, bool) and maybe_code_list is False:
+#             return None
+#
+#         else:
+#             raise Exception(f"Unexpected codelist value '{maybe_code_list}'")
+#
+#     return NewQbCodeListInCsvW(
+#         json_parent_dir
+#         / "codelists"
+#         / f"{uri_safe(column_label)}.csv-metadata.json"
+#     )
+
+
+def initialise_from_config(self, config_json: dict) -> bool:
+    """
+    Initialises a Qube from a qube json file.
+
+    param: config_json: A Qube created according to a Qube Config schema
+    """
+
+
+def _override_catalog_metadata_state(
+    catalog_metadata_json_file: Path, cube: QbCube
+) -> None:
+    with open(catalog_metadata_json_file, "r") as f:
+        catalog_metadata_dict: dict = json.load(f)
+    overriding_catalog_metadata = CatalogMetadata.from_dict(catalog_metadata_dict)
+    cube.metadata.override_with(
+        overriding_catalog_metadata,
+        overriding_keys={k for k in catalog_metadata_dict.keys()},
+    )
+
+
+def _from_config_json_dict(d: Dict,
+                           data: pd.DataFrame,
+                           json_parent_dir: Path
+                          ) -> QbCube:
+
+    metadata: CatalogMetadata = _metadata_from_dict(d)
+    # columns: List[CsvColumn] = _columns_from_config_json(d.get("columns", []), data, json_parent_dir)
+    columns: List[QbColumn] = []
+    for column_title in d.get('columns', []):
+        column_config = d['columns'].get(column_title)
+        columns.append(map_column_to_qb_component(column_title, column_config,
+                                                  data[column_title], json_parent_dir))
+
+    return Cube(metadata, data, columns)
+
+
+def _override_config_for_cube_id(config: dict, cube_id: str) -> Optional[dict]:
```

Comment:
We should get rid of this now. It isn't necessary for this qube-config.json stuff.

Reply:
Commented out, then deleted in later commit.